pcap/pcap.go

```diff
@@ -455,6 +456,7 @@ func (p *Handle) compileBPFFilter(expr string) (_Ctype_struct_bpf_program, error
 // CompileBPFFilter compiles and returns a BPF filter for the pcap handle.
 func (p *Handle) CompileBPFFilter(expr string) ([]BPFInstruction, error) {
 	bpf, err := p.compileBPFFilter(expr)
+	defer C.pcap_freecode(&bpf)
```

Comment:
Also, the compileBPFFilter call is not exported, so library users don't have to care too much :)